### PR TITLE
[WIP] MSI with azure go client.

### PR DESCRIPTION
**Pull request by kwoodson**

### CODE REVIEW

File: pkg/addons/addons.go [hunk 1]
```diff
@@ -57,9 +62,23 @@ func readDB(cs *acsapi.ContainerService, c *config.Config) (map[string]unstructu
 			return nil, err
 		}
 
+		azs, err := GetStorageAccountsClient(c)
```
Comment (kargakis) [inline pkg/addons/addons.go]: run this outside of the loop?

Comment (kargakis) [reply pkg/addons/addons.go]: Or maybe move this code altogether in newClient. I think it makes more sense in there.

Comment (kwoodson) [reply pkg/addons/addons.go]: Where do you recommend this go?  As part of the client instantiation?  As a var inside of the addons.go?  Inside of addons.go main?

Comment (kargakis) [reply pkg/addons/addons.go]: it's client-related code so client instantiation would be my choice.

File: pkg/addons/addons.go [hunk 2]
```diff
@@ -57,9 +62,23 @@ func readDB(cs *acsapi.ContainerService, c *config.Config) (map[string]unstructu
...
+		}
+
+		ex := &extra{
+			StorageAccountKey: *(((*response.Keys)[0]).Value),
+		}
+
 		ts := Translations[KeyFunc(o.GroupVersionKind().GroupKind(), o.GetNamespace(), o.GetName())]
 		for _, tr := range ts {
-			b, err := util.Template(tr.Template, nil, cs, c, nil)
+			b, err := util.Template(tr.Template, nil, cs, c, ex)
```
Comment (mjudeikis) [inline pkg/addons/addons.go]: Would it not be better to set it as c *Config object, so it will get persisted together with all config?

**Comment by kwoodson** [comment]
Closing.  #144 was merged.

**Comment by kargakis** [comment]
@kwoodson can you make sure we track this work in Jira? Thanks!

**Pull request closed without merging**